torch/_dynamo/variables/tensor.py

```diff
@@ -222,35 +222,56 @@ def var_getattr(self, tx, name):
             if name in self._strict_mode_banned_ops():
                 unimplemented(f"Illegal getattr invocation {name} in strict mode")
 
-        result = None
-        if name == "ndim" and self.ndim is not None:
-            result = ConstantVariable.create(self.ndim)
-        elif name == "dtype" and self.dtype is not None:
-            result = ConstantVariable.create(self.dtype)
-        elif name == "device" and self.device is not None:
-            result = ConstantVariable.create(self.device)
-        elif name == "layout" and self.layout is not None:
-            result = ConstantVariable.create(self.layout)
-        elif name == "is_cuda" and self.device is not None:
-            result = ConstantVariable.create(self.device.type == "cuda")
-        elif name == "shape" and self.size is not None:
-            sizes = [variables.ConstantVariable.create(x) for x in self.size]
-            result = SizeVariable(sizes)
-        elif name == "requires_grad" and self.requires_grad is not None:
-            result = ConstantVariable.create(self.requires_grad)
-        elif name == "is_quantized" and self.is_quantized is not None:
-            result = ConstantVariable.create(self.is_quantized)
-        elif name == "is_sparse" and self.is_sparse is not None:
-            result = ConstantVariable.create(self.is_sparse)
-        elif name == "shape" and self.size is None:
-            result = self.call_method(tx, "size", [], {})
-        elif name == "ndim" and self.ndim is None:
-            result = self.call_method(tx, "dim", [], {})
-        elif name == "data":
-            result = self.call_method(tx, "detach", [], {})
         if name == "__class__":
             return UserDefinedClassVariable(self.python_type())
 
+        def method_ndim():
```

Comment:
I do worry that refactors like this will make the code harder to read/debug

Reply:
I think this is ok. We have similar handler for builtins. So it should improve readability in the long run.